### PR TITLE
Replace localhost to real ip address

### PR DESCRIPTION
**Pull request by yanyzy**
Complete TODO of replace localhost to real ip address for MySQLAuthenticationEngine

### CODE REVIEW

File: sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java [hunk 1]
```diff
@@ -60,9 +63,10 @@ public boolean auth(final ChannelHandlerContext context, final PacketPayload pay
             backendConnection.setUserName(response41.getUsername());
             context.writeAndFlush(new MySQLOKPacket(response41.getSequenceId() + 1));
         } else {
-            // TODO localhost should replace to real ip address
+            SocketAddress socketAddress = context.channel().remoteAddress();
+            String hostAddress = socketAddress instanceof InetSocketAddress ? ((InetSocketAddress) socketAddress).getAddress().getHostAddress() : socketAddress.toString();
```
Comment (tuohai666) [inline sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java]: What if the socketAddress is null? You can find the null value possibility in the comment of API.

Comment (yanyzy) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java]: I find “null if this channel is not connected” in the API comment, but this method is called by auth method, i think the channel is connected

Comment (tuohai666) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java]: Yes, you are right.

**Comment by terrymanu** [comment]
Thanks for contribution

**Comment by grissomsh** [comment]
Great!

**Pull request merged**